homeassistant/components/mychevy.py

```diff
@@ -0,0 +1,133 @@
+"""
+MyChevy Component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/mychevy/
+"""
+
+from datetime import timedelta
+import logging
+import time
+import threading
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ["mychevy==0.1.1"]
+
+DOMAIN = 'mychevy'
+UPDATE_TOPIC = DOMAIN
+ERROR_TOPIC = DOMAIN + "_error"
+
+MYCHEVY_SUCCESS = "success"
+MYCHEVY_ERROR = "error"
+
+NOTIFICATION_ID = 'mychevy_website_notification'
+NOTIFICATION_TITLE = 'MyChevy website status'
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+ERROR_SLEEP_TIME = timedelta(minutes=30)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+class EVSensorConfig(object):
+    """EV Sensor Config."""
+
+    def __init__(self, name, attr, unit_of_measurement=None, icon=None):
+        """Create new Sensor Config."""
+        self.name = name
+        self.attr = attr
+        self.unit_of_measurement = unit_of_measurement
+        self.icon = icon
+
+
+class EVBinarySensorConfig(object):
+    """EV Binary Sensor Config."""
+
+    def __init__(self, name, attr, device_class=None):
+        """Create new Binary Sensor Config."""
+        self.name = name
+        self.attr = attr
+        self.device_class = device_class
+
+
+def setup(hass, base_config):
+    """Setup mychevy platform."""
+    import mychevy.mychevy as mc
+
+    config = base_config.get(DOMAIN)
+    _LOGGER.debug('Received configuration: %s', config)
```

Comment:
I don't think you should log the credentials.

Reply:
ack